lib/chai/core/assertions.js

```diff
@@ -988,6 +988,12 @@ module.exports = function (chai, _) {
 
   function assertInstanceOf (constructor, msg) {
     if (msg) flag(this, 'message', msg);
+    var constructorType = constructor === null ? 'null' : typeof constructor;
```

Comment:
More concise option

```js
  function assertInstanceOf (constructor, msg) {
    if (msg) flag(this, 'message', msg);
    var validInstanceOfTarget = constructor === Object(constructor) && (
        typeof constructor == 'function' ||
        typeof constructor[Symbol.hasInstance] == 'function'
    );
    if (!validInstanceOfTarget) throw new TypeError();

    var name = _.getName(constructor);
    this.assert(
        flag(this, 'object') instanceof constructor
      , 'expected #{this} to be an instance of ' + name
      , 'expected #{this} to not be an instance of ' + name
    );
  };
```

But this one calls `constructor.[[Get]](@@hasInstance)` twice. `[[Get]]` may be observable.